text/0000-unify-katello-ui.md

```diff
@@ -0,0 +1,78 @@
+- Feature Name: Unify Katello UI
+- Start Date: 2016-08-26
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The goal of this feature is to combine [Katello](github.com/katello/katello) and [Bastion](github.com/katello/bastion)
+into a single UI project.
+
+# Motivation
+[motivation]: #motivation
+
+Currently when working on the UI in Katello a developer has to oftentimes make changes in both the Katello and Bastion
+projects in order to complete a feature.  Once Katello begins to make more use of the components from
+[angular-patternfly](https://github.com/patternfly/angular-patternfly) this problem will only become worse as there will
+be three potential UI repositories to alter in order to complete new features.
+
+The expected outcome of this feature is to reduce the complexity of working on UI code in Katello by combining projects.
+
+# Detailed design
+[design]: #detailed-design
+
+## Option 1:  Move Bastion code back into Katello
+
+Eliminate Bastion entirely and move everything in Bastion back into Katello.
+
+### Pros
+- One less repository to manage
+- Would no longer require the release of a Bastion gem to update UI code in Katello
+- UI changes would be easier because they would only be made in one repository/project
+
+### Cons
+- Decreases separation of concerns, the UI would be bundled with the API code (although you could count this as a pro
+since it would bring us into consistency with [Foreman](https://github.com/theforeman/foreman))
+
+## Option 2:  Move Katello UI code into Bastion
+
+Keep Bastion and move all AngularJS based Katello UI code into Bastion
+
+### Pros
+- UI code and API code is separated
+- Potential to remove rails from the UI code
+
+### Cons
+- Still have two repositories/projects to manage
+- Would have to release the UI code to see changes in the main project
+
+## Option 3:  Move API code into an API project, combine Bastion back into Katello
```

Comment:
Yeah, I guess that is true for the most part.

The main difference is that in option 3 the code and configuration for displaying web pages to the browser would be in the UI project whereas the API would be only things needed for an API and AR and it would be able to stand alone from the UI.

I think in option 2 the katello project would consist of the API plus some other UI specific code or configuration.  I may be wrong about this and you'll have to pardon my lack of knowledge about engine configuration maybe @ehelms or someone else can chime in here.

Other than that the project names would match up better in option 3 in my opinion and it may be easier pull out the API from katello than it is to pull out the UI, I am not sure.

I can combine these two options into a 2 a,b on the next revision of this RFC if needed.